src/core_cli/Core_CLI.ml

```diff
@@ -276,6 +291,9 @@ let all_actions () =
     ( "-dump_pattern",
       " <file>",
       Arg_helpers.mk_action_1_conv Fpath.v dump_pattern );
+    ( "-dump_patterns_of_rule",
```

Comment:
I think it would be better instead to use -dump_rule and just force the lazy eval of the pattern before, so the dump looks good. I'm not sure what you refer to as "non sense" in the PR, but whatever I'm fine with an extra action if it helps people.

Reply:
one thing was that, because `dump_rule` uses the rule printing function `Rule.show`, it doesn't do the nice things which `dump_pattern` and `vof_any` do, which is hide some of the irrelevant details. that also motivated this PR a bit